pkg/base1/cockpit.js

```diff
@@ -153,30 +153,54 @@ function transport_debug() {
 }
 
 function event_mixin(obj, handlers) {
-    obj.addEventListener = function addEventListener(type, handler) {
-        if (handlers[type] === undefined)
-            handlers[type] = [ ];
-        handlers[type].push(handler);
-    };
-    obj.removeEventListener = function removeEventListener(type, handler) {
-        var length = handlers[type] ? handlers[type].length : 0;
-        for (var i = 0; i < length; i++) {
-            if (handlers[type][i] === handler) {
-                handlers[type][i] = null;
-                break;
+    Object.defineProperties(obj, {
+        addEventListener: {
+            enumerable: false,
+            value: function addEventListener(type, handler) {
+                if (handlers[type] === undefined)
+                    handlers[type] = [ ];
+                handlers[type].push(handler);
+            }
+        },
+        removeEventListener: {
+            enumerable: false,
+            value: function removeEventListener(type, handler) {
+                var length = handlers[type] ? handlers[type].length : 0;
```

Comment:
since we don't have to worry about concurrency, why not use something like this here:
```
var index = handlers[type] ? handlers[type].indexOf(handler) : -1;
if (index > -1)
    handlers[type].splice(index, 1);
```
That seems clearer and we don't have null values cluttering the array.